%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
\RequirePackage{hyperref}
\usepackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Tools and pipelines for BioNano data: molecule assembly pipeline and FASTA super scaffolding tool}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   %noteref={n1},                        % id's of article notes, if any
   email={sheltonj@ksu.edu}   % email address
]{\inits{JM}\fnm{Jennifer M} \snm{Shelton}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={cmgordon@ksu.edu}   % email address
]{\inits{MC}\fnm{Michelle C} \snm{Coleman}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={nherndon@ksu.edu}   % email address
]{\inits{N}\fnm{Nic} \snm{Herndon}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={nanyan@ksu.edu}   % email address
]{\inits{N}\fnm{Nanyan} \snm{Lu}}
\author[
   addressref={aff1},
      corref={aff1},                       % id of corresponding address, if any
   email={sjbrown@ksu.edu}
]{\inits{SJ}\fnm{Susan J} \snm{Brown}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Biology, Kansas State University}, % university, etc
  %\street{Waterloo Road},                     %
  %\postcode{}                                % post or zip code
  \city{Manhattan},                              % city
  \cny{KS, USA}                                    % country
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{Background} %if any
Text for this section.
\parttitle{Results} %if any
In this article we present software that leverages BioNano genome maps assembled from ultra-long single molecule maps to improve the contiguity of sequence assemblies. We report the results of applying these tools to a 7x Sanger draft of the \textit{Tribolium castaneum} genome.
\parttitle{Conclusions} %if any
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Genome map}
\kwd{BioNano}
\kwd{Genome scaffolding}
\kwd{Genome validation}
\kwd{Genome finishing}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
The quality of genome assemblies varies greatly. Initial assembly drafts, whether based on lower coverage Sanger or higher coverage NGS reads, are often highly fragmented. Physical maps of BAC clones are often use to validate and scaffold sequence assemblies, but the molecular, human and computational resources required to significantly improve a draft genome are not often available to researchers working on non-model organisms. The BioNano Irys platform provides affordable, high throughput physical maps with significantly higher contiguity with which to validate draft assemblies and extend scaffolds.

\subsection*{Data formats}
The tools described make use of three file formats developed by BNG. The Irys platform images ultra-long molecules of genomic DNA that are nick-labeled at 7 (bp) motifs using one or more nicking endonucleases and fluorescently labeled nucleotides. Molecules captured in the TIFF images are converted to BNX format text files that include label position for each molecule (steps 1 and 2 Figure 1). These BNX files are referred to as molecule maps. Consensus Map (CMAP) files include the molecule length and label position for long genomic regions that are either inferred from assembly of BNX molecules (steps 7 and 8 figure 1) or \textit{in silico} from sequence scaffolds (steps 3 and 4 figure 1). These two types of CMAPs are referred to as BioNano genome maps and \textit{in silico} genome maps respectively. The alignment of two CMAPs is stored as an XMAP text file that includes alignment coordinates and an alignment confidence score (step 10 Figure 1).

\section*{Implementation}
\subsection*{Assembly preparation}
We have developed AssembleIrysCluster to prepare BNX files for assembly and produce nine customized assembly scripts (sections C-G Figure 2).
\subsubsection*{Molecule Stretch}
In the first stage AssembleIrysCluster adjusts molecule stretch (section C Figure 2). BioNano software opperates under the assumption that imaged molecules contain 500 bases per pixel (bpp). Stretch, or bpp, can deviate from 500 bpp and this discrepancy can vary from scan to scan within a flowcell (ADD BPP GRAPH TO SUPPLEMENTARY FILES). Sequence scaffolds are considered to be more accurate than raw BNX molecules in terms of label position. Therefore molecule maps in a BNX files are split by scan number and after alignment to the \textit{in silico} genome map an empirical bpp value is determined for the molecule maps in each scan. The bpp indicated by this alignment is used by RefAligner to adjust molecule map bpp to 500. Once stretch has been evaluated and adjusted the split BNX files are merged into a single file (section E Figure 2).

We obseved consitant patterns of empirically determined bpp between flowcells using the same flow cell model and chemistry when signal-to-noise ratios are optimal and the degree of genomic divergence between the samples used for the \textit{in silico} genome map and molecule maps are low. (ADD BPP GRAPH TO SUPPLEMENTARY FILES HIGHLIGHT DIVERGENT AND SNR ISSUES VS SUCCESSFUL ALIGNMENT). To identify the low quality flowcells bpp observed in alignments of scans are plotted as a QC graph (section D Figure 2). 

\subsubsection*{Customization of Assembly Scripts}

In the next stage of AssembleIrysCluster creates various assembly scripts are created to explore a range of parameters with the goal of selecting the optimal assembly for downstream analysis (sections F-G Figure 2). Molecule maps in the merged adjusted BNX file are aligned to the \textit{in silico} genome map. An alignment error profile generated by RefAligner is used with the estimated genome length to calculate default assembly parameters and the eight other scripts include variants of these. Initially three assemblies are run with $p\textnormal{-}Value Threshold Default = \frac{1e-5}{Genome Length (Mb)}$, $p\textnormal{-}Value Threshold Strict = \frac{p\textnormal{-}Value Threshold}{10}$ and $p\textnormal{-}Value Threshold Relaxed = {p\textnormal{-}Value Threshold} \times {10}$ (section F Figure 2). Minimum molecule length is set to 150 kb. If these do not produce a satisfactory assembly then two minimum molecule length variants (180 kb and 100 kb) are tested with the $p\textnormal{-}Value Threshold$ of the current best assembly (section G Figure 2). Between three and nine assemblies are run until an assembly is produced that is satisfactory.

\subsubsection*{Assembly Optimization}

Ultimately the goal is to produce a CMAP that can be used to guide sequence-based haploid reference genome assembly. While BioNano genome maps can be used to reconstruct haplotypes \cite{MHCLam} genome assembly involves collapsing polymorphisms arbitrarily into a consensus reference genome. Therefore for an ideal BioNano genome map $Length (Mb) = Genome Length (Mb)$. Additionally, 100\% of the BioNano genome map would align non-redundantly to 100\% of the \textit{in silico} genome map. When BioNano genome maps are imperfect optimal assembly length is balanced against alignment quality when selecting the best CMAP.

\subsubsection*{Stitch: Alignment Filters}

Alignments, XMAPs, of the \textit{in silico} genome maps and the BioNano genome maps are used the predict the higher order arrangement of genome scaffolds with Stitch (Figure 3). RefAligner is designed to treat the reference genome map as an \textit{in silico} genome map and the query as the BioNano genome map. Therefore alignment XMAPs are first inverted and sorted by BioNano genome map coordinates for efficient parsing by Stitch (section A and B Figure 3). 

Before inferring super scaffolds from XMAPs, Stitch filters low quality alignments by confidence score. Alignments of \textit{in silico} and BioNano genome maps are assigned a confidence score that is the $-log10$ of the $False\textnormal{ }Positive\textnormal{ }p\textnormal{-}Value$. Misaligned labels and sizing error increase the alignment $False\textnormal{ }Positive\textnormal{ }p\textnormal{-}Value$ decreasing confidence scores \cite{ProbFP} .

Alignments are also filtered by the percent of the total possible alignment length that is aligned (section C Figure 3). Super scaffolds are built from overlapping alignments. Overlapping alignments are similar to global alignments, i.e. alignments spanning from end to end for two maps of roughly equal length, but to search for overlap alignments gaps after the ends of either map are not penalized. The RefAligner scoring scheme does not currently have a parameter to favor overlapping alignments, e.g. to initialize the dynamic programming matrix with no penalties and take the maximum score of the final row or column in the matrix. Refaligner reports local alignments between two maps and applies a fixed penalty based on the user defined likelihood of unaligned labels at the ends of an alignment. Raising or lowering this penalty selects for local or global alignments respectively but neither option favors overlapping alignments specifically. To approximate scoring that favors overlapping alignments Stitch uses thresholds for minimum percent of total possible aligned length or the percent aligned threshold (PAT). 

Like a scoring structure that favors overlapping alignments the PAT filters out local alignments. However, unlike a scoring structure the PAT is applied after alignment and therefore cannot result in the aligner exploring possible extensions into an overlap in favor of a shorter local alignment with a higher cumulative score. Therefore Stitch accepts alignments with less than 100\% of the potential aligned length. Default values for the PAT were determined empirically after reviewing the degree to which filtered alignments agreed with the independent genetic maps of \textit{Tribolium} and visual inspection.

In practice it was found useful to use two sets of alignment filters and keep alignments that passed one or both sets. The first set has a lower PAT and a higher confidence score threshold. The second set has a high PAT and a lower confidence score and is intended for lower label density regions of the genome.

\subsubsection*{Stitch: Super Scaffolding}

Scaffolding alignments are next selected from the the remaining high quality alignments (i.e. more than one \textit{in silico} genome map aligns to the same BioNano genome map). From these the best alignment is selected for each \textit{in silico} genome map with more than one passing and scaffolding alignment. The best alignment is considered to be the longest alignment in base pairs. If alignment length is identical than the highest confidence alignment is selected. If confidence scores are identical than an alignment is chosen arbitrarily.

The gap lengths between \textit{in silico} genome maps are inferred from scaffolding alignments and used to create a new super scaffolded sequence FASTA file and an AGP file. If gap lengths are estimated to be negative Stitch adds a 100 bp spacer gap to the sequence file and indicates that the gap is type "U" for unknown in the AGP. 

Stitch only makes use of one alignment per \textit{in silico} map per iteration. Stitch can be run iteratively so that each successive output FASTA file is \textit{in silico} nicked and the new \textit{in silico} genome map is aligned to the original BioNano genome map. This alignment is inverted and used as input for the next iteration. Subsequent iterations of Stitch will make use of any \textit{in silico} genome maps that join growing super scaffolds effectively using both sequence data and genome maps to stitch together the final super scaffolds. 

\subsubsection*{Stitch: Flagging Potential Mis-assemblies}

This algorithm is meant to be an intermediate refinement of draft genomes prior to further fine scale refinement at the sequence level. Inconsistencies between the BioNano genome maps and the \textit{in silico} genome maps are reported in output logs to facilitate downstream sequence editing. If an alignment passes initial confidence score and PAT filtering but has a PAT less then 60\% this is reported as a partial alignment. A partial alignment may occur if either the sequence scaffold or the BioNano genome map contained a chimeric assembly. Additionally, if a gap length is estimated to be negative it may indicate that the sequence scaffolds can be joined with a local assembly or that a chimeric sequence mis-assembly needs to be broken within a scaffold. Assembly errors in the BioNano genome maps or spurious alignments could also result in either of these cases. Ideally researchers could make use of the alignment of genomic sequence reads to the genome sequence assembly and the alignment of BioNano molecule maps to the BioNano assembled genome map to determine which assembly is likely to be incorrect.
 
\section*{Results and Discussion}

\subsubsection*{Datasets}

For 7x draft reference 2008 paper...

The reference genome scaffolds are from version 5.0 of the \textit{Tribolium castaneum} genome. Between version 3.0 and version 5.0 of the \textit{Tribolium castaneum} genome 1,025,047 bases were added and 2,221 scaffolds remained after scaffold linking with Atlas-Link (version 0.01) and gap filling with GapFiller (version 1.10). The version 5.0 assembly has 2240 scaffolds with a cumulative length of 160.74 Mb and an N50 of 1.16 Mb (Table 1).

Young pupae from an inbred Ga-I line of \textit{Tribolium castaneum} (inbred 20 generations) was harvested and 0.5 g was vacuum cleaned through a seive to remove contaminants. Pupae were ground in liquid nitrogen with a cryogenic mortar. After filtration and differential centifugation clean nuclei were embedded in agarose plugs. Subsequent protein digestion and wash steps included gelase and dialysis to clean DNA. DNA was nick, label repaired with nt.BspQI and nt.BbvCI from NEB and run in 96 flowcells on the Iyrs genome mapping platform. 

The 563,140 molecule maps with lengths of equal to or longer than 100kb had an N50 of 165.35 kb and a cummulative length of 827,738.48 Mb. Molecule maps with a minimum length of 150 kb had... More detailed quality metrics for molecule maps after appling a minimum molecule map filter of either 100 kb or 150 kb are reported in Supplemental\_x\_MapStatsHistograms\_min\_100kb and Supplemental\_x\_MapStatsHistograms\_min\_150kb respectively.




\section*{Conclusions}
Text


\section*{Availability and requirements}

\subsection*{\textbf{Assembly scripts}}

\textbf{Project name:} AssembleIrysCluster.pl

\textbf{Project home page:} AssembleIrysCluster scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/assemble\_SGE\_cluster}

\textbf{Operating system(s):} SGE Linux (tested on a Gentoo) cluster

\textbf{Programming language:} Perl, Rscript, Bash

\textbf{License}: AssembleIrysCluster.pl is available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} AssembleIrysCluster.pl requires DRMAA job submission libraries. RefAligner and Assembler are also required and can be provided by request by Bionano Genomics \url{http://www.bionanogenomics.com/}.

\subsection*{\textbf{Super scaffolding scripts}}

\textbf{Project name:} stitch.pl

\textbf{Project home page:} stitch scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/stitch}

\textbf{Operating system(s):} MAC and LINUX (tested on Gentoo and Ubuntu)

\textbf{Programming language:} Perl, Bash

\textbf{License}: stitch.pl is available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} stitch.pl requires BioPerl. RefAligner and Assembler are also required between iterations and can be provided by request by Bionano Genomics \url{http://www.bionanogenomics.com/}.

\subsection*{\textbf{Map summary scripts}}

\textbf{Project name:} cmap\_stats.pl and xmap\_stats.pl

\textbf{Project home page:} all scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/map\_editing}

\textbf{Operating system(s):} MAC and LINUX (tested on Gentoo and Ubuntu)

\textbf{Programming language:} Perl

\textbf{License}: cmap\_stats.pl and xmap\_stats.pl are available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} cmap\_stats.pl and xmap\_stats.pl have no dependencies.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  Text for this section \ldots
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  	\caption{\csentence{Data analysis steps.}
  		(1) Autonoise converts TIFF images of molecules to (2) BNX text files. (3) Sequence scaffolds are \textit{in silico} labeled with fa2cmap\_multi producing (4) a \textit{in silico} genome map. (5) AssembleIryscluster uses \textit{in silico} genome maps, BNX files and estimated genome size to (6) adjust molecule stretch and set assembly parameters. (7) Assembler produces (8) a BioNano genome map. (9) RefAligner aligns the BioNano genome map to the \textit{in silico} genome map producing (10) an XMAP. (11) XMAP, \textit{in silico} genome map and BioNano genome map (see arrows with dashed lines) are used by stitch to produce super scaffolded (stitched) sequence scaffolds. (13) Until no more super scaffolds are created the stitched sequence scaffolds are \textit{in silico} labeled with fa2cmap\_multi producing (14) a CMAP that is aligned to (9) the BioNano genome map and steps 10-15 are iterated. Arrows with dotted rather than dashed lines are used to as input durring iterations.}
  \end{figure}
  \begin{figure}[h!]
  	\caption{\csentence{Assembly workflow for assemble\_SGE\_cluster.pl.}
  		(A) The Irys instrument produces tiff files that are converted into BNX text files. (B) Each chip produces one BNX file for each of two flowcells. (C) BNX files are split by scan and aligned to the sequence reference. Stretch (bases per pixel) is recalculated from the alignment. (D) Quality check graphs are created for each pre-adjusted flowcell BNX. (E) Adjusted flowcell BNXs are merged. (F) The first assemblies are run with a variety of p-value thresholds. (G) The best of the first assemblies (red oval) is chosen and a version of this assembly is produced with a variety of minimum molecule length filters.}
  \end{figure}
  \begin{figure}[h!]
  \caption{\csentence{Steps of the stitch.pl algorithm.}
      BioNano genome maps (blue) are shown aligned to \textit{in silico} genome maps (green). Alignments are indicated with grey lines. CMAP orientation for \textit{in silico} genome maps is indicated with a "+" or "-" for positive or negative orientation respectively. (A) The \textit{in silico} genome map is aligned as the reference. (B) The alignment is inverted and used as input for stitch.pl. (C) The alignments are filtered based on alignment length (purple) relative to total possible alignment length (black) and confidence. Here assuming all alignments have a high confidence score and the minimum percent aligned is 30\% two alignments fail for aligning over less than 30\% of the potential alignment length for that alignment. (D) Filtering produces an XMAP of high quality alignments with short (local) alignments removed. (E) High quality scaffolding alignments are filtered for longest and highest confidence alignment for each \textit{in silico} genome map. Third alignment (unshaded) is filtered because the second alignment is the longest alignment for {in silico} genome map 2. (F) Passing alignments are used to super scaffold (captured gaps indicated in dark green). (G) Stitch is iterated and additional super scaffolding alignments are found using second best scaffolding alignments. (H) Iteration takes advantage of cases where \textit{in silico} genome maps scaffold BioNano genome maps as \textit{in silico} genome map 2 does. Stitch is run iteratively all until super scaffolding alignments are found.}
      \end{figure}      
\begin{figure}[h!]
	\caption{\csentence{Putative haplotypes assembled as BioNano genome maps.}
		(A) Two BioNano genome maps (blue with molecule coverage shown in dark blue) align to the \textit{in silico} genome map of scaffold 131 (green with contigs overlaid as translucent colored squares). (B and C) Both BioNano genome maps are shown (blue) with molecule pileups (yellow). Both BioNano genome maps have similar label patterns except within the lower coverage region indicated with a black square.}
\end{figure}  
\begin{figure}[h!]
	\caption{\csentence{Histogram of gap lengths in Tcas5.1.}
		Positive and negative gaps lengths for Tcas5.1 added to the automated output of stitch.pl based on filtered scaffolding alignments. The majority of gap lengths added by stitch.pl, 66, were positive (red). The remaining 26 gaps had negative lengths (purple).}
      \end{figure}                      
\begin{figure}[h!]
	\caption{\csentence{Extremely small negative gap length for \textit{in silico} genome map of scaffold 81.}
		Two XMAP alignments for \textit{in silico} genome map of sequence scaffold 81 are shown. BioNano genome maps (blue with molecule coverage shown in dark blue) align to the \textit{in silico} genome maps of scaffolds (green with contigs overlaid as translucent colored squares). Sequence scaffolds 79-83 were placed within ChLG 5 and sequence scaffolds 99-103 were placed with ChLG 7 by the \textit{Tribolium} genetic map. (A) Half of the \textit{in silico} genome map of sequence scaffold 81 aligns with its assigned ChLG (black arrow). (B) The other half aligns with ChLG 7 (red arrow) producing a negative gap length smaller than -20 kb. The alignment that places sequence scaffold 81 with ChLG 7 disagrees with the genetic map and was manually rejected for Tcas5.2.}
      \end{figure}            


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%

\section*{Tables}
\begin{table}[h!]
	\caption{Assembly Results. Assembly metrics for Tcas5.0 (the starting sequence scaffolds), the \textit{in silico} genome map, the BioNano genome map of assembled molecules and the final super scaffolded sequence scaffolds (Tcas5.2) produced using stitch.pl for the \textit{Tribolium} genome.}
	\begin{tabular}{cccc}
		\hline
		& N50 (Mb)  & Number & Cumulative Length (Mb)\\ \hline
		Sequence scaffolds & 1.16 & 2240 & 160.74\\
		\textit{in silico} genome map & 1.20 &  223 & 152.53\\
		BioNano genome map & 1.35 &  216 & 200.47\\
		Super scaffolded sequence scaffolds & 4.46 & 2150 & 165.92\\ \hline
	\end{tabular}
\end{table}
\begin{table}[h!]
\caption{Alignment of BNG assembly to reference genome. Breadth of alignment coverage (non-redundant alignment), length of total alignment (including redundant alignments) and percent of CMAP covered (non-redundantly) were calculated for the \textit{in silico} genome map and the BioNano genome map of the \textit{Tribolium} genome the using xmap\_stats.pl.}
      \begin{tabular}{cccc}
        \hline
           & Breadth of alignment coverage (Mb) & Length of total alignment (Mb) & Percent of CMAP aligned \\ \hline
        \textit{in silico} genome map from FASTA & 124.04 & 132.40 & 81 \\
        BioNano genome map & 131.64 & 132.34 & 67 \\ \hline
      \end{tabular}
\end{table}
\begin{table}[h!]
	\caption{Chromosome linkage groups before and after super scaffolding. The number of sequence scaffolds in the Tcas5.0 ChLG bins and the number of sequence super scaffolds and scaffolds in the Tcas5.2 ChLG bins. The number of sequence scaffolds  that were unplaced in Tcas5.0 and placed with a ChLG in Tcas5.2 is also listed. }
	\begin{tabular}{cccc}
		\hline
		Chromosome linkage group (ChLG) & Tcas5.0 sequence scaffolds & Unplaced sequence scaffolds added in Tcas5.2 & Tcas5.2 sequence super scaffolds \\ \hline
		X & 13 & +2 & 2\\
		2 & 18 & +1 & 10\\
		3 & 29 & +4 & 20\\
		4 & 6 & +2 & 2\\
		5 & 17 & +1 & 4\\
		6 & 12 & +6 & 6\\
		7 & 15 & - & 6\\
		8 & 14 & +1 & 8\\
		9 & 21 & - & 9\\
		10 & 12 & +2 & 11\\ \hline
		Total & 157 & 78 & 19\\ \hline
	\end{tabular}
\end{table}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
