%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
\RequirePackage{hyperref}
\usepackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Tools and pipelines for BioNano data: molecule assembly pipeline and FASTA super scaffolding tool}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   %noteref={n1},                        % id's of article notes, if any
   email={sheltonj@ksu.edu}   % email address
]{\inits{JM}\fnm{Jennifer M} \snm{Shelton}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={cmgordon@ksu.edu}   % email address
]{\inits{MC}\fnm{Michelle C} \snm{Coleman}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={nherndon@ksu.edu}   % email address
]{\inits{N}\fnm{Nic} \snm{Herndon}}
\author[
addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
%noteref={n1},                        % id's of article notes, if any
email={nanyan@ksu.edu}   % email address
]{\inits{N}\fnm{Nanyan} \snm{Lu}}
\author[
   addressref={aff1},
      corref={aff1},                       % id of corresponding address, if any
   email={sjbrown@ksu.edu}
]{\inits{SJ}\fnm{Susan J} \snm{Brown}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Biology, Kansas State University}, % university, etc
  %\street{Waterloo Road},                     %
  %\postcode{}                                % post or zip code
  \city{Manhattan},                              % city
  \cny{KS, USA}                                    % country
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{First part title} %if any
Text for this section.

\parttitle{Second part title} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Genome map}
\kwd{Bionano}
\kwd{Genome scaffolding}
\kwd{Genome validation}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
In this article we present software that leverages genome maps assembled from ultra-long single molecule maps to improve the contiguity of the NGS assemblies. We report the results of applying these tools to the \textit{Tribolium castaneum} genome.

\subsection*{Data formats}
The tools described make use of three file formats developed by BNG. The Irys platform images ultra-long molecules of genomic DNA that are nick-labeled at 7 (bp) motifs using one or more nicking endonucleases and fluorescently labeled nucleotides. Molecules captured in the TIFF images are converted to BNX format text files that include label position for each molecule (steps 1 and 2 Figure 1). Consensus Map (CMAP) files include the length and label position for long genomic regions that are either inferred from assembly of BNX molecules (steps 7 and 8 figure 1) or from \textit{in silico} labeling of a NGS scaffolds (steps 3 and 4 figure 1). The two types of CMAPs are referred to as BioNano genome maps and \textit{in silico} genome maps respectively. The alignment of two CMAPs is stored as an XMAP text file which includes alignment coordinates and an alignment confidence score (step 10 Figure 1).

\section*{Implementation}
\subsection*{Assembly preparation}
We have developed AssembleIrysCluster to prepare BNX files for assembly and write nine customized assembly scripts (sections C-G Figure 2).
\subsubsection*{Molecule Stretch}
The first stage of AssembleIrysCluster is to adjust molecule stretch (section C Figure 2). Imaged molecules are presumed to have 500 bases per pixel (bpp). Stretch, or bpp, can deviate from 500 bpp and this discrepancy can vary from scan to scan for a flowcell (ADD BPP GRAPH TO SUPPLEMENTARY FILES). NGS scaffolds are considered to be more accurate than raw BNX molecules in terms of label position. Therefore BNX files are split by scan number, aligned to the \textit{in silico} genome map and a empirical bpp value is determined. The bpp indicated by this alignment is used to adjust BNX bpp to 500. Optimal average flow cell signal-to-noise ratio and a low degree of genomic divergence between the samples used for the \textit{in silico} genome map and BNX molecules have been associated with a characteristic pattern of empirically determined bpp when using the most recent flow cell model and chemistry (ADD BPP GRAPH TO SUPPLEMENTARY FILES HIGHLIGHT DIVERGENT AND SNR ISSUES VS SUCCESSFUL ALIGNMENT). Therefore bpp observed in alignments of scans are plotted as a QC graph (section D Figure 2). Once stretch has been evaluated and adjusted the split BNX files are merged (section E Figure 2).

\subsubsection*{Customization of Assembly Scripts}

In the next stage of AssembleIrysCluster various assembly scripts are created to explore a range of parameters with the goal of selecting the optimal assembly for downstream analysis (sections F-G Figure 2). The merged adjusted BNX file is aligned to the \textit{in silico} genome map. The alignment error profile is used with the estimated genome length to calculate default assembly parameters and the eight other scripts include variants of these. Initially three assemblies are run with $p-Value Threshold = \frac{1e-5}{Genome Length (Mb)}$, $p-Value Threshold Strict = \frac{p-Value Threshold}{10}$ and $p-Value Threshold Relaxed = {p-Value Threshold} \times {10}$ (section F Figure 2). Minimum molecule length is set to 150 kb. If these do not produce a satisfactory assembly than two minimum molecule length variants (180 kb and 100 kb) are tested with the $p-Value Threshold$ of the current best assembly (section G Figure 2). Ultimately up to nine assemblies are run until an assembly is produced that is satisfactory.

\subsubsection*{Assembly Optimization}

Ultimately the goal is to produce a CMAP that can be used to guide sequence-based haploid reference genome assembly. While BioNano genome maps can be used to reconstruct haplotypes \cite{MHCLam} genome assembly involves collapsing polymorphisms arbitrarily into a consensus reference genome. Therefore for an ideal BioNano genome map $Length (Mb) = Genome Length (Mb)$. Additionally, 100\% of the BioNano genome map would align non-redundantly to 100\% of the \textit{in silico} genome map. When BioNano genome maps are imperfect optimal assembly length is balanced against alignment quality when selecting the best CMAP.

\subsubsection*{Super scaffolding}
Text
 
\section*{Results and Discussion}
Text

\section*{Conclusions}
Text


\section*{Availability and requirements}

\subsection*{\textbf{Assembly scripts}}

\textbf{Project name:} AssembleIrysCluster.pl

\textbf{Project home page:} AssembleIrysCluster scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/assemble\_SGE\_cluster}

\textbf{Operating system(s):} SGE Linux (Gentoo) cluster

\textbf{Programming language:} Perl, Rscript, Bash

\textbf{License}: AssembleIrysCluster.pl is available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} AssembleIrysCluster.pl requires DRMAA job submission libraries. RefAligner and Assembler are also required and can be provided by request by Bionano Genomics \url{http://www.bionanogenomics.com/}.

\subsection*{\textbf{Super scaffolding scripts}}

\textbf{Project name:} stitch.pl

\textbf{Project home page:} stitch scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/stitch}

\textbf{Operating system(s):} MAC and LINUX (tested on Gentoo and Ubuntu)

\textbf{Programming language:} Perl, Bash

\textbf{License}: stitch.pl is available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} stitch.pl requires BioPerl. RefAligner and Assembler are also required between iterations and can be provided by request by Bionano Genomics \url{http://www.bionanogenomics.com/}.

\subsection*{\textbf{Map summary scripts}}

\textbf{Project name:} cmap\_stats.pl and xmap\_stats.pl

\textbf{Project home page:} all scripts are available on Github at \url{https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/tree/master/KSU\_bioinfo\_lab/map\_editing}

\textbf{Operating system(s):} MAC and LINUX (tested on Gentoo and Ubuntu)

\textbf{Programming language:} Perl

\textbf{License}: cmap\_stats.pl and xmap\_stats.pl are available free of charge to academic and non-profit institutions.

\textbf{Any restrictions to use by non-academics:} Please contact authors for commercial use.

\textbf{Dependencies:} cmap\_stats.pl and xmap\_stats.pl have no dependencies.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  Text for this section \ldots
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  	\caption{\csentence{Data analysis steps.}
  		(1) Autonoise converts TIFF images of molecules to (2) BNX text files. (3) NGS scaffolds are \textit{in silico} labeled with fa2cmap\_multi producing (4) a \textit{in silico} genome map. (5) AssembleIryscluster uses \textit{in silico} genome maps, BNX files and estimated genome size to (6) adjust molecule stretch and set assembly parameters. (7) Assembler produces (8) a BioNano genome map. (9) RefAligner aligns the BioNano genome map to the \textit{in silico} genome map producing (10) an XMAP. (11) XMAP, \textit{in silico} genome map and BioNano genome map are used by stitch to produce super scaffolded NGS scaffolds. (13) Until no more super scaffolds are created the super scaffolded NGS scaffolds are \textit{in silico} labeled with fa2cmap\_multi producing (14) a CMAP that is aligned to (9) the BioNano genome map and steps 10-15 are iterated.}
  \end{figure}
  \begin{figure}[h!]
  	\caption{\csentence{Assembly workflow for assemble\_SGE\_cluster.pl.}
  		(A) The Irys instrument produces tiff files that are converted into BNX text files. (B) Each chip produces one BNX file for each of two flowcells. (C) BNX files are split by scan and aligned to the sequence reference. Stretch (bases per pixel) is recalculated from the alignment. (D) Quality check graphs are created for each pre-adjusted flowcell BNX. (E) Adjusted flowcell BNXs are merged. (F) The first assemblies are run with a variety of p-value thresholds. (G) The best of the first assemblies (red oval) is chosen and a version of this assembly is produced with a variety of minimum molecule length filters.}
  \end{figure}
  \begin{figure}[h!]
  \caption{\csentence{Steps of the stitch.pl algorithm.}
      BioNano genome maps (blue) are shown aligned to \textit{in silico} genome maps (green). Alignments are indicated with grey lines. CMAP orientation for \textit{in silico} genome maps is indicated with a "+" or "-" for positive or negative orientation respectively. (A) The \textit{in silico} genome map is aligned as the reference. (B) The alignment is inverted and used as input for stitch.pl. (C) The alignments are filtered based on alignment length (purple) relative to total possible alignment length (black) and confidence. Here assuming all alignments have a high confidence score and the minimum percent aligned is 30\% two alignments fail for aligning over less than 30\% of the potential alignment length for that alignment. (D) Filtering produces an XMAP of high quality alignments with short (local) alignments removed. (E) High quality scaffolding alignments are filtered for longest and highest confidence alignment for each \textit{in silico} genome map. Third alignment (unshaded) is filtered because the second alignment is the longest alignment for {in silico} CMAP 2. (F) Passing alignments are used to super scaffold (captured gaps indicated in dark green). (G) Stitch is iterated and additional super scaffolding alignments are found using second best scaffolding alignments. (H) Iteration takes advantage of cases where \textit{in silico} genome maps scaffold BioNano genome maps as \textit{in silico} genome map 2 does. Stitch is run iteratively until super scaffolding alignments are found.}
      \end{figure}      
\begin{figure}[h!]
	\caption{\csentence{Putative haplotypes assembled as BioNano genome maps.}
		(A) Two BioNano genome maps (blue with molecule coverage shown in dark blue) align to the \textit{in silico} genome map of scaffold 131 (green with contigs overlaid as translucent colored squares). (B and C) Both BioNano genome maps are shown (blue) with molecule pileups (yellow). Both BioNano genome maps have similar label patterns except within the lower coverage region indicated with a black square.}
\end{figure}  
\begin{figure}[h!]
	\caption{\csentence{Histogram of gap lengths in Tcas5.1.}
		Positive and negative gaps lengths for Tcas5.1 added to the automated output of stitch.pl based on filtered scaffolding alignments. The majority of gap lengths added by stitch.pl, 66, were positive (red). The remaining 26 gaps had negative lengths (purple).}
      \end{figure}                      
\begin{figure}[h!]
	\caption{\csentence{Extremely small negative gap length for \textit{in silico} genome map of scaffold 81.}
		Two XMAP alignments for \textit{in silico} genome map of NGS scaffold 81 are shown. BioNano genome maps (blue with molecule coverage shown in dark blue) align to the \textit{in silico} genome maps of scaffolds (green with contigs overlaid as translucent colored squares). NGS scaffolds 79-83 were placed within ChLG 5 and NGS scaffolds 99-103 were placed with ChLG 7 by the \textit{Tribolium} genetic map. (A) Half of the \textit{in silico} genome map of NGS scaffold 81 aligns with its assigned ChLG (black arrow). (B) The other half aligns with ChLG 7 (red arrow) producing a negative gap length smaller than -20 kb. The alignment that places NGS scaffold 81 with ChLG 7 disagrees with the genetic map and was manually rejected for Tcas5.2.}
      \end{figure}            


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%

\section*{Tables}
\begin{table}[h!]
	\caption{Assembly Results. Assembly metrics for Tcas5.0 (the starting NGS scaffolds), the \textit{in silico} genome map, the BioNano genome map of assembled molecules and the final super scaffolded NGS scaffolds (Tcas5.2) produced using stitch.pl for the \textit{Tribolium} genome.}
	\begin{tabular}{cccc}
		\hline
		& N50 (Mb)  & Number & Cumulative Length (Mb)\\ \hline
		NGS scaffolds & 1.16 & 2240 & 160.74\\
		\textit{in silico} genome map & 1.20 &  223 & 152.53\\
		BioNano genome map & 1.35 &  216 & 200.47\\
		Super scaffolded NGS scaffolds & 4.46 & 2150 & 165.92\\ \hline
	\end{tabular}
\end{table}
\begin{table}[h!]
\caption{Alignment of BNG assembly to reference genome. Breadth of alignment coverage (non-redundant alignment), length of total alignment (including redundant alignments) and percent of CMAP covered (non-redundantly) were calculated for the \textit{in silico} genome map and the BioNano genome map of the \textit{Tribolium} genome the using xmap\_stats.pl.}
      \begin{tabular}{cccc}
        \hline
           & Breadth of alignment coverage (Mb) & Length of total alignment (Mb) & Percent of CMAP aligned \\ \hline
        \textit{in silico} genome map from FASTA & 124.04 & 132.40 & 81 \\
        BioNano genome map & 131.64 & 132.34 & 67 \\ \hline
      \end{tabular}
\end{table}
\begin{table}[h!]
	\caption{Chromosome linkage groups before and after super scaffolding. The number of NGS scaffolds in the Tcas5.0 ChLG bins and the number of NGS super scaffolds and scaffolds in the Tcas5.2 ChLG bins. The number of NGS scaffolds  that were unplaced in Tcas5.0 and placed with a ChLG in Tcas5.2 is also listed. }
	\begin{tabular}{cccc}
		\hline
		Chromosome linkage group (ChLG) & Tcas5.0 NGS scaffolds  & Tcas5.2 NGS super scaffolds & Unplaced NGS scaffolds added in Tcas5.2\\ \hline
		X & 13 & 2 & 2\\
		2 & 18 & 10 & 1\\
		3 & 29 & 20 & 4\\
		4 & 6 & 2 & 2\\
		5 & 17 & 4 & 1\\
		6 & 12 & 6 & 6\\
		7 & 15 & 6 & 0\\
		8 & 14 & 8 & 1\\
		9 & 21 & 9 & 0\\
		10 & 12 & 11 & 2\\ \hline
		Total & 157 & 78 & 19\\ \hline
	\end{tabular}
\end{table}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
