stitch_paper
============

#Tools and pipelines for BioNano data: molecule assembly pipeline and FASTA super scaffolding tool

###Jennifer M Shelton, Michelle C Coleman, Nic Herndon, Nanyan Lu, Ernest T Lam, Thomas Anantharaman, Palak Sheth and Susan J Brown

Abstract
Background: Genome assembly remains an unsolved problem. Assembly projects face a range of hurdles that confound assembly. Thus a variety of tools and approaches are needed to improve draft genomes.
Results: We used a custom assembly workflow to optimize consensus genome map assembly, resulting in an assembly equal to the estimated length of the Tribolium castaneum genome and with an N50 of more than 1 Mb. We used this map for super scaffolding the T. castaneum sequence assembly, more than tripling its N50 with the program Stitch.
Conclusions: In this article we present software that leverages consensus genome maps assembled from extremely long single molecule maps to increase the contiguity of sequence assemblies. We report the results of applying these tools to validate and improve a 7x Sanger draft of the T. castaneum genome.

Keywords: Genome map; BioNano; Genome scaffolding; Genome validation; Genome finishing


Download PDF from [bmc_article.pdf](https://github.com/kstatebioinfo/stitch_paper/raw/master/bmc_article/bmc_article.pdf)

View Figures at:

[Fig_1_overall_workflow.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_1_overall_workflow.pdf)

[Fig_2_bionano_assembly_workflow.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_2_bionano_assembly_workflow.pdf)

[Fig_3_detailed_stitch_steps.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_3_detailed_stitch_steps.pdf)

[Fig_4_flowcell_date.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_4_flowcell_date.pdf)

[Fig_5_assembly_metrics.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_5_assembly_metrics.pdf)

[Fig_6_gap_length_histogram.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_6_gap_length_histogram.pdf)

[Fig_7_xmaps_scaffold_81.png](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_7_xmaps_scaffold_81.png):

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/Fig_7_xmaps_scaffold_81.png)

[Fig_8_ChLGX_and_hybrid_map.pdf](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_8_ChLGX_and_hybrid_map.pdf)

[Fig_9_putative_haplotype.png](https://github.com/kstatebioinfo/stitch_paper/blob/master/figures/Fig_9_putative_haplotype.png):

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/Fig_9_putative_haplotype.png)

Chromosome Linkage groups before and after:

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLGX_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG2_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG3_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG4_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG5_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG6_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG7_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG8_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG9_before_and_after.png)

![Alt text](https://github.com/kstatebioinfo/stitch_paper/raw/master/figures/ChLG10_before_and_after.png)

